### PR TITLE
Add achievement definitions

### LINKED ISSUE #1058: Create achievement definitions and write a function to check and assign them to users
Badge Name | Task
-------|-----
In Pursuit of Mappiness | Followed 5 Groups
Ready, Set, Roll | Finished Online Training
Certified Mapper | Attended a Training Event
Free Mapper | Attended a Mapping Event
Mapped 50 Block Segments | Mapped 50 Block Segments
Mapped 100 Block Segments | Mapped 100 Block Segments
Mapped 200 Block Segments | Mapped 200 Block Segments
Mapped 400 Block Segments | Mapped 400 Block Segments
Mapped 1000 Block Segments | Mapped 1000 Block Segments

### PR DESCRIPTION
**Pull request by RickMohr**
* Create achievement definitions per #1058
* `update_achievements()` creates `Achievement` objects for specified user
* Test each kind of achievement

Also create helpers `complete_online_training` and `blocks_mapped_count`

Fixes #1058

### CODE REVIEW

**Comment by kdeloach** [comment]
+1 Nice.

File: src/nyc_trees/apps/users/models.py [hunk 1]
```diff
@@ -15,45 +15,124 @@
...
+    achieved - a function which takes a user and returns a boolean
     """
-    def __init__(self, name, description, badge, progress):
+    def __init__(self, name, description, description_future, badge, achieved):
         assert isinstance(name, basestring)
         assert isinstance(description, basestring)
+        assert isinstance(description_future, basestring)
         assert isinstance(badge, basestring)
-        assert hasattr(progress, '__call__')
+        assert hasattr(achieved, '__call__')
```
Comment (steventlamb) [inline src/nyc_trees/apps/users/models.py]: I think this should be
`assert callable(achieved)`

File: src/nyc_trees/apps/users/models.py [hunk 2]
```diff
@@ -15,45 +15,124 @@
...
         self.name = name
         self.description = description
+        self.description_future = description_future
         self.badge = badge
-        self.progress = progress
+        self.achieved = achieved
 
-    FINISH_TRAINING = 0
-    JOIN_10_GROUPS = 1
+    FOLLOW_GROUPS = 0
```
Comment (steventlamb) [inline src/nyc_trees/apps/users/models.py]: We present `groups to follow` after online training is complete, should these be reversed?

**Comment by steventlamb** [comment]
+1 on review, looks nice. Will test tomorrow.

**Comment by steventlamb** [comment]
I spun up with this branch and visited my uper profile and my achievements section has no content:
screenshot from 2015-04-14 11 11 05

Do we want to add something here?

**Comment by maurizi** [comment]
@steventlamb I would assume your user hasn't achieved anything yet.  That said, we should probably add some content there when you have no acheivements (maybe just a link to the main achievements page?)

File: src/nyc_trees/apps/users/models.py [hunk 3]
```diff
@@ -15,45 +15,124 @@
 class AchievementDefinition(object):
     """A class representing an achievment definition
     name - string - title of the achievement
-    description - string - description of the achievement requirements
+    description - requirements (after achievement)
```
Comment (steventlamb) [inline src/nyc_trees/apps/users/models.py]: I find these names confusing. Consider `description_complete` and `description_pending`.

**Comment by steventlamb** [comment]
tests pass

File: src/nyc_trees/apps/users/models.py (hunk 3 above)
Comment (kdeloach) [inline src/nyc_trees/apps/users/models.py]: I vote for `description` (present tense) and `description_past` (past tense).

File: src/nyc_trees/apps/users/models.py (hunk 1 above)
Comment (RickMohr) [inline src/nyc_trees/apps/users/models.py]: done.

File: src/nyc_trees/apps/users/models.py (hunk 2 above)
Comment (RickMohr) [inline src/nyc_trees/apps/users/models.py]: done.

File: src/nyc_trees/apps/users/models.py (hunk 3 above)
Comment (RickMohr) [inline src/nyc_trees/apps/users/models.py]: I used Kevin's suggestion.

**Pull request merged**